lib/bolt/plugin_cache.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'bolt/error'
+require 'bolt/util'
+
+module Bolt
+  class PluginCache
```

Comment:
Small thing, but I think this should be a nested class of `Bolt::Plugin`, since its behavior is specific to plugins and it already has 'Plugin' in its name.

Reply:
All the current plugin subclasses are plugins themselves, so I think that'd make it seem like this is a caching plugin and not a builtin feature.

Reply:
`Bolt::Plugin::Module` isn't a plugin itself 🤷